File: src/main/kotlin/problems/FoxAndSnake.kt
package problems

/* Problem Link: https://codeforces.com/problemset/problem/510/A */

fun main() {
    val (n, m) = readln().split(" ").map { it.toInt() }
    var start = true

    for (i in 1..n) {
        if (i % 2 != 0) start = !start

        for (j in 1..m) {
            if (i % 2 != 0) {
                print('#')
            } else if (j != 1 && j != m) print('.')
            else if (start && j == 1) print('#')
            else if (!start && j == m) print('#')
            else print('.')
        }

        println()
    }
}

File: Codeforces/C++/problemset/Fox And Snake.cpp
// Problem Link: https://codeforces.com/problemset/problem/510/A

#include <iostream>
using namespace std;
typedef long long ll;
int main()
{
    ll n, m;
    cin >> n >> m;
    for(ll i = 0; i < n; i++)
    {
        for(ll j = 0; j < m; j++)
        {
            if(i%2 == 0) cout << "#";
            else
            {
                if(i%4 == 1 && j == m-1) cout <<  "#";
                else if(i%4 == 3 && j == 0) cout << "#";
                else cout << ".";
            }
        }
        cout << "\n";
    }
    return 0;
}
